CodeGeneration/Sources/SyntaxSupport/KeywordSpec.swift

```diff
@@ -13,34 +13,69 @@
 import SwiftSyntax
 
 public struct KeywordSpec {
-  public var name: String
+  /// The name of the keyword.
+  public let name: String
 
-  /// If `true`, this is for an experimental language feature, and any public
-  /// API generated should be SPI.
-  public var isExperimental: Bool
+  /// Indicates if the keyword is part of an experimental language feature.
+  ///
+  /// If `true`, this keyword is for an experimental language feature, and any public
+  /// API generated should be marked as SPI
+  public let isExperimental: Bool
 
-  public var isLexerClassified: Bool
-  public var requiresLeadingSpace: Bool
-  public var requiresTrailingSpace: Bool
+  /// Indicates if the token kind is switched from being an identifier to a keyword in the lexer.
+  public let isLexerClassified: Bool
 
-  public var escapedName: String {
+  /// Indicates if the keyword requires a leading space.
+  ///
+  /// If `true`, the keyword should have a space character before it when used in Swift code.
+  public let requiresLeadingSpace: Bool
+
+  /// Indicates if the keyword requires a trailing space.
+  ///
+  /// If `true`, the keyword should have a space character after it when used in Swift code.
+  public let requiresTrailingSpace: Bool
+
+  /// The escaped name of the keyword.
+  ///
+  /// This is useful when the keyword is also an identifier or has special characters that need escaping.
+  public var escapedName: TokenSyntax {
     if isLexerClassified || name == "Type" || name == "Protocol" {
-      return "`\(name)`"
+      return "`\(raw: name)`"
     } else {
-      return name
+      return "\(raw: name)"
     }
   }
 
-  /// Retrieve the attributes that should be printed on any API for the
-  /// generated keyword.
+  /// The name of the keyword enclosed in double quotes.
+  ///
+  /// This property is used in code generation where a string representation of the keyword is needed.
+  public var quotedName: TokenSyntax {
+    "\"\(raw: name)\""
+  }
```

Comment:
This isn’t correct because you’re returning a string literal here, which, in SwiftSyntax, is represented by a [StringLiteralExprSyntax](https://swiftpackageindex.com/apple/swift-syntax/main/documentation/swiftsyntax/stringliteralexprsyntax) node, and not a single token. What I would suggest is that in the code that access is, you do `\(literal: keyword.name)`, which would also add pounds in case the name would contain `"` or anything else that needed escaping (not that I would ever expect that from a keyword name but it’s good to be safe).

Reply:
Oh right! My bad.

> What I would suggest is that in the code that access is, you do `\(literal: keyword.name)`, which would also add pounds in case the name would contain `"` or anything else that needed escaping (not that I would ever expect that from a keyword name but it’s good to be safe).

But it would change generated source code, right? Do we mind such a change?

Reply:
Oh right! My bad! Sorry for that. I've removed this property and I've used `literal:` as you suggested. Great idea, thanks so much!